src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/MemoryUsageSummary.java

```diff
@@ -0,0 +1,189 @@
+ *
+ */
+package org.rstudio.studio.client.workbench.views.environment.view;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.dom.client.Document;
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.dom.client.TableCellElement;
+import com.google.gwt.dom.client.TableRowElement;
+import com.google.gwt.resources.client.CssResource;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.HTMLPanel;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.Widget;
+import org.rstudio.core.client.ElementIds;
+import org.rstudio.core.client.StringUtil;
+import org.rstudio.core.client.widget.MiniPieWidget;
+import org.rstudio.studio.client.workbench.views.environment.model.MemoryStat;
+import org.rstudio.studio.client.workbench.views.environment.model.MemoryUsage;
+import org.rstudio.studio.client.workbench.views.environment.model.MemoryUsageReport;
+
+public class MemoryUsageSummary extends Composite
+{
+   interface Style extends CssResource
+   {
+      String kbCell();
+      String stats();
+   }
+
+   /**
+    * Creates a new memory usage summary widget
+    *
+    * @param report The memory usage report to show in the widget.
+    */
+   public MemoryUsageSummary(MemoryUsageReport report)
+   {
+      report_ = report;
+
+      initWidget(GWT.<MemoryUsageSummary.Binder>create(MemoryUsageSummary.Binder.class).createAndBindUi(this));
+
+      MemoryUsage usage = report.getSystemUsage();
+
+      // Create the pie chart showing the percentage of memory used
+      int percent = usage.getPercentUsed();
+      pie_.setForeColor(usage.getColorCode());
+      pie_.setBackColor(MemUsageWidget.MEMORY_PIE_UNUSED_COLOR);
+      pie_.setPercent(percent);
+      pie_.getElement().getStyle().setWidth(150, com.google.gwt.dom.client.Style.Unit.PX);
+      pie_.getElement().getStyle().setHeight(150, com.google.gwt.dom.client.Style.Unit.PX);
+      pie_.setTitle("Memory in use: " + usage.getPercentUsed());
+      pie_.setDescription("Pie chart depicting the percentage of total memory in use");
+      ElementIds.assignElementId(pie_, ElementIds.MEMORY_PIE_FULL);
+      pieLabel_.setText(percent + "%");
+
+      Element statsTable = Document.get().createTableElement();
+      ElementIds.assignElementId(statsTable, ElementIds.MEMORY_USAGE_TABLE);
+      statsTable.setClassName(style.stats());
+
+      // Create the header for the table:
+      //
+      // Statistic  Memory  Source
+      // ---------  ------  ------
+      Element statsHeader = Document.get().createTHeadElement();
+      Element statsRow = Document.get().createTRElement();
+      Element statCell = Document.get().createTDElement();
+      statCell.setInnerText("Statistic");
+      statsRow.appendChild(statCell);
+      Element memoryCell = Document.get().createTDElement();
```

Comment:
`statsCell`, `memoryCell`, and `sourceCell` should be TH elements, not TD elements, otherwise the table isn't correctly formed for a screen reader to recognize that these are headings. This is a check that isn't performed by the automated tools, unfortunately, so we have this problem all over the place (ditto for unnamed tables).

Reply:
These are now `<th>` elements.